pkg/operator/ceph/file/mds/spec.go

```diff
@@ -127,7 +128,19 @@ func (c *Cluster) makeChownInitContainer(mdsConfig *mdsConfig) v1.Container {
 	)
 }
 
-func (c *Cluster) makeMdsDaemonContainer(mdsConfig *mdsConfig) v1.Container {
+func (c *Cluster) getCephConfig() (string, string) {
+	monHost, monIntialMemebers := "", ""
+	cephConfigSecret, err := c.context.Clientset.CoreV1().Secrets(c.clusterInfo.Namespace).Get(c.clusterInfo.Context, "rook-ceph-config", metav1.GetOptions{})
+	if err != nil {
+		logger.Info("rook-ceph-config secret is not available")
+	} else {
+		monHost = string(cephConfigSecret.Data["mon_host"])
+		monIntialMemebers = string(cephConfigSecret.Data["mon_initial_members"])
+	}
+	return monHost, monIntialMemebers
+}
```

Comment:
Rook should not be setting the liveness probe's mon connection info like this. This effectively hard-codes the connection info to a specific point-in-time value that can become invalid after mon failovers.

Reply:
Changed